src/server/auth/cmds/cmds.go

```diff
@@ -492,27 +508,46 @@ func UseAuthTokenCmd() *cobra.Command {
 
 // GetOneTimePasswordCmd returns a cobra command that lets a user get an OTP.
 func GetOneTimePasswordCmd() *cobra.Command {
+	var ttl string
 	getOneTimePassword := &cobra.Command{
-		Use:   "{{alias}} <username>",
-		Short: "Get a one-time password that authenticates the holder as \"username\"",
-		Long:  "Get a one-time password that authenticates the holder as \"username\"",
-		Run: cmdutil.RunFixedArgs(1, func(args []string) error {
-			subject := args[0]
+		Use: "{{alias}} <username>",
+		Short: "Get a one-time password that authenticates the holder as " +
+			"\"username\", or the currently signed in user if no 'username' is " +
+			"specified",
+		Long: "Get a one-time password that authenticates the holder as " +
+			"\"username\", or the currently signed in user if no 'username' is " +
+			"specified. Only cluster admins may obtain a one-time password on " +
+			"behalf of another user.",
+		Run: cmdutil.RunBoundedArgs(0, 1, func(args []string) error {
 			c, err := client.NewOnUserMachine("user")
 			if err != nil {
 				return fmt.Errorf("could not connect: %v", err)
 			}
 			defer c.Close()
-			resp, err := c.GetOneTimePassword(c.Ctx(), &auth.GetOneTimePasswordRequest{
-				Subject: subject,
-			})
+
+			req := &auth.GetOneTimePasswordRequest{}
+			if ttl != "" {
+				d, err := time.ParseDuration(ttl)
+				if err != nil {
+					return fmt.Errorf("could not parse duration %q: %v", ttl, err)
+				}
+				req.TTL = int64(d.Seconds())
+			}
+			if len(args) == 1 {
+				req.Subject = args[0]
+			}
+			resp, err := c.GetOneTimePassword(c.Ctx(), req)
 			if err != nil {
 				return grpcutil.ScrubGRPC(err)
 			}
 			fmt.Println(resp.Code)
 			return nil
 		}),
 	}
+	getOneTimePassword.PersistentFlags().StringVar(&ttl, "ttl", "", "if set, "+
+		"the resulting one-time password will have the given lifetime (or the "+
+		"lifetime of the caller's current session, whichever is shorter). This "+
+		"flag should be a golang duration (e.g. \"30s\" or \"1d2h3m4s\")")
```

Comment:
likewise, what happens if this is not set?

Reply:
Updated description